packages/openid4vc/src/shared/utils.ts

```diff
@@ -124,13 +134,20 @@ export async function openIdTokenIssuerToJwtIssuer(
       certificateChain: openId4VcTokenIssuer.x5c,
     })
 
+    const jwk = getJwkFromKey(leafCertificate.publicKey)
+    const alg = jwk.supportedSignatureAlgorithms[0]
+    if (!alg) {
+      throw new CredoError(`No supported signature algorithms found key type: '${jwk.keyType}'`)
+    }
+
     if (!issuer.startsWith('https://')) {
       throw new CredoError('The X509 certificate issuer must be a HTTPS URI.')
     }
 
     if (leafCertificate.sanUriNames?.includes(issuer)) {
       return {
         ...openId4VcTokenIssuer,
+        alg: alg as unknown as SigningAlgo,
```

Comment:
I think the lib should just support string. As you can use any algorithm and this is a bit annoying to cast everytime